src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -2188,6 +2320,32 @@ private StackEntry NewSpillSlot(StackEntry entry)
             }
         }
 
+        private StackEntry TakeAddressOf(StackEntry entry)
+        {
+            var entryType = entry.Type ?? GetWellKnownType(WellKnownType.Object); //type is required here, currently the only time entry.Type is null is if someone has pushed a null literal
+
+            LLVMValueRef addressValue;
+            if(entry is LoadExpressionEntry)
+            {
+                addressValue = ((LoadExpressionEntry)entry).RawLLVMValue;
+            }
+            else
+            {
+                var entryIndex = _spilledExpressions.Count;
```

Comment:
Does this end up duplicating value types and returning the address of the temp? That's ok for boxing, but it's a problem for something like 
```
ldloca someStruct
ldflda someStruct.someFld
ldc.0
stind.i
```

Reply:
It should not result in duplicating value types. ldloca/ldflda/ldelema/ret will always produce a LoadExpressionEntry to prevent duplicates from occurring in your described situation.

Reply:
There is something wrong here, once I figure it out, I'll include a test.

Reply:
There was an unrelated issue with ldind when I wrote the test for this, should I include that fix in this PR or make a separate one.

Reply:
I think this change is big enough already, just file an issue or send a separate PR.